caip-browser_extension_provider_communication.md

```diff
@@ -0,0 +1,137 @@
+* receive messages by using `port.onMessage.addListener()`
+  * incoming messages should be validated according to the message format specification above
+
+The webpage embedded provider can:
+* initiate a connection with the web extension by using `port = browser.runtime.connect()`
+* send messages by using `port.postMessage()`
+* receive messages by using `port.onMessage.addListener()`
+
+### Caveats
+Currently Firefox does not support `externally_connectable` yet, but they are [considering implementing it](https://bugzilla.mozilla.org/show_bug.cgi?id=1319168). Meanwhile, extension wallets on Firefox will need to continue injecting an inpage provider for the dApp.
+
+## Rationale
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+While web extensions could realize the benefits of `externally_connectable` without using it with a standardized interface, this would mean that every web extension would have to ship a provider library that worked specifically for their interface and that Dapps would need to import them. It wouldn't be feasible for Dapps to import every single web extension's specific provider implementation, making this approach less viable.
+
+Web extension inter-operability is the key to enabling generalized provider implementations. Generalized provider implementations allows for convenient adoption by Dapps, leading to more wide spread adoption of the standard as a result.
+
+## Test Cases
+<!--Please add test cases here if applicable.-->
+
+## Security Considerations
+<!--Please add an explicit list of intra-actor assumptions and known risk factors if applicable. Any normative definition of an interface requires these to be implementable; assumptions and risks should be at both individual interaction/use-case scale and systemically, should the interface specified gain ecosystem-namespace adoption. -->
+`externally_connectable` has seen a decade of usage via extensions on Chrome. It has a strictly better security when compared to postMessage over contentscript.
+
+## Privacy Considerations
+<!--Please add an explicit list of intra-actor assumptions and known risk factors if applicable. Any normative definition of an interface requires these to be implementable; assumptions and risks should be at both individual interaction/use-case scale and systemically, should the interface specified gain ecosystem-namespace adoption. -->
+Exposing the wallet API over `externally_connectable` opens a migration path towards a bring-your-own provider model in which web extensions can reduce their fingerprint by no longer having to inject their own provider into every webpage.
```

Comment:
Probably worth expanding on the privacy considerations of the discovery mechanisms, when that is documented somewhere. That's likely where we'll have more to say.

Though... couldn't `externally_connectable` be used for fingerprinting? I guess it would be no worse than the fingerprinting that would be possible today with injected code or a contentscript that replies to certain messages - someone could reach out to us in any context and use that to help identify a person. But perhaps we should explain this risk and ensure that it's characterized as no worse than the status quo.